src/libstd/sys/windows/mutex.rs

```diff
@@ -157,34 +157,37 @@ fn kind() -> Kind {
     return ret;
 }
 
-pub struct ReentrantMutex { inner: UnsafeCell<c::CRITICAL_SECTION> }
+pub struct ReentrantMutex { inner: MaybeUninit<UnsafeCell<c::CRITICAL_SECTION>> }
```

Comment:
Would it not make more sense to keep `MaybeUninit` *within* `UsafeCell`?

Reply:
Good point, fixed.